components/remote_settings/src/storage.rs

```diff
@@ -140,30 +176,13 @@ impl Storage {
         }
     }
 
-    /// Set the list of records stored in the database, clearing out any previously stored records
-    pub fn set_records(
-        &mut self,
-        collection_url: &str,
-        records: &[RemoteSettingsRecord],
-    ) -> Result<()> {
-        let tx = self.conn.transaction()?;
-
-        tx.execute("DELETE FROM records", [])?;
-        tx.execute("DELETE FROM collection_metadata", [])?;
-        let max_last_modified = Self::update_record_rows(&tx, collection_url, records)?;
-        Self::update_collection_metadata(&tx, collection_url, max_last_modified)?;
-        tx.commit()?;
-        Ok(())
-    }
-
-    /// Merge new records with records stored in the database
-    ///
-    /// Records with `deleted=false` will be inserted into the DB, replacing any previously stored
-    /// records with the same ID. Records with `deleted=true` will be removed.
-    pub fn merge_records(
+    /// Set cached content for this collection.
+    pub fn insert_collection_content(
         &mut self,
         collection_url: &str,
         records: &[RemoteSettingsRecord],
+        timestamp: u64,
```

Comment:
Nit: how about `last_modified` or `last_modified_timestamp`?  `timestamp` could also mean the timestamp when we fetched the data.